packages/itwin/ec3-widget/src/components/DeleteModal.tsx

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import {
+  Button,
+  Leading,
+  MiddleTextTruncation,
+  Modal,
+  ModalButtonBar,
+} from "@itwin/itwinui-react";
+import React, { useState } from "react";
+import "./DeleteModal.scss";
+import { handleError, LoadingSpinner } from "./utils";
+
+export interface DeleteModalProps {
+  entityName: string;
+  show: boolean;
+  setShow: React.Dispatch<React.SetStateAction<boolean>>;
+  onDelete: () => Promise<void>;
+  refresh: () => Promise<void>;
+}
+
+export const DeleteModal = ({
+  entityName,
+  show,
+  setShow,
+  onDelete,
+  refresh,
+}: DeleteModalProps) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const deleteCallback = async () => {
+    try {
+      setIsLoading(true);
+      await onDelete();
+      setShow(false);
+      await refresh();
+    } catch (error: any) {
+      handleError(error.status);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <>
```

Comment:
The empty fragment component is not required - you are returning a single `Modal` component.